cmds/core/cat/cat_test.go

```diff
@@ -59,3 +68,53 @@ func TestCat(t *testing.T) {
 		t.Fatalf("Reading files failed: got %v, want %v", b.Bytes(), someData)
 	}
 }
+
+func TestCatPipe(t *testing.T) {
+	var inputbuf bytes.Buffer
+	teststring := "testdata"
+	fmt.Fprintf(&inputbuf, "%s", teststring)
+	osStdin = &inputbuf
+
+	var out bytes.Buffer
+	osStdout = &out
+
+	emptyargs := make([]string, 0)
+	if err := cat(&out, emptyargs); err != nil {
+		t.Error(err)
+	}
+	if out.String() != teststring {
+		t.Errorf("Input string: %s and output: '%s' string are not equal", teststring, out.String())
+	}
+	resetGlobals()
+}
+
+func TestFileNotExist(t *testing.T) {
+	args := make([]string, 0)
+	filepath := "testdata/doesnotexist.txt"
+	args = append(args, filepath)
+	var b bytes.Buffer
+	if err := cat(&b, args); !errors.Is(err, os.ErrNotExist) {
+		t.Error(err)
+	}
+	t.Log(b.String())
+}
+
+func TestIOCopyFail(t *testing.T) {
+	var inputbuf bytes.Buffer
+	teststring := "testdata"
+	fmt.Fprintf(&inputbuf, "%s", teststring)
+	osStdin = &inputbuf
+
+	var out bytes.Buffer
+	osStdout = &out
+
+	ioCopy = func(out io.Writer, in io.Reader) (int64, error) {
+		return 0, fmt.Errorf("io copy fail")
+	}
+
+	emptyargs := make([]string, 0)
+	if err := cat(&out, emptyargs); !strings.Contains(err.Error(), "io copy fail") {
+		t.Error(err)
+	}
+	resetGlobals()
```

Comment:
This is very racy. the resetGlobals of one test might somehow run right after another test has set them. In general, use of this kind of global is a danger sign that things must be done differently. I suggest making the cat function as I described above; and see how some of our other tests run the program and gather the output.